middleware/lib/FunctionInvocationRequest.ts

```diff
@@ -0,0 +1,87 @@
+
+import {Logger} from '@salesforce/core/lib/logger';
+import {SObject} from '@salesforce/salesforce-sdk/dist/objects';
+import {
+    DataApi,
+    ErrorResult,
+    SuccessResult,
+} from '@salesforce/salesforce-sdk/dist/api';
+
+export enum FunctionInvocationRequestStatusEnum {
+  Success = 'Success',
+  Error = 'Error'
+}
+
+// Save update to FunctionInvocationRequest
+// Does not throw
+export async function saveFnInvocation(logger: Logger,
+                                       fnInvocation: FunctionInvocationRequest,
+                                       response: any,
```

Comment:
@jabrown85 comment was lost on the latest push.

Chatted on Slack.  In a nutshell, we'll monitor whether a developer returns an unresolved `Promise` w/in the returned userFn resolved `Promise`.